airflow/jobs/dag_processor_job_runner.py

```diff
@@ -53,6 +53,7 @@ def __init__(
                 f"The job is already assigned a different job_type: {job.job_type}."
                 f"This is a bug and should be reported."
             )
+        self.job.job_type = self.job_type
```

Comment:
Should this be done in the case class so no other job subclasses forget it?

Reply:
Yeah. Possible